src/formatDate.js

```diff
@@ -1,14 +1,54 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prevSplitter = fromFormat[3];
+  const newSplitter = toFormat[3];
+  const rawDate = date.split(prevSplitter);
+  const newRawDate = [];
+
+  function setTheDate(dateType, nextDateType = dateType) {
+    newRawDate[toFormat.indexOf(nextDateType)] =
+      rawDate[fromFormat.indexOf(dateType)];
+  }
+
+  setTheDate('DD');
+  setTheDate('MM');
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YYYY')) {
+    setTheDate('YYYY');
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    setTheDate('YYYY', 'YY');
+
+    newRawDate[toFormat.indexOf('YY')] = newRawDate[
+      toFormat.indexOf('YY')
+    ].slice(2, 4);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (rawDate[fromFormat.indexOf('YY')] < 30) {
+      newRawDate[toFormat.indexOf('YYYY')] =
+        `20${rawDate[fromFormat.indexOf('YY')]}`;
+    }
+
+    if (rawDate[fromFormat.indexOf('YY')] >= 30) {
+      newRawDate[toFormat.indexOf('YYYY')] =
+        `19${rawDate[fromFormat.indexOf('YY')]}`;
+    }
+  }
```

Comment:
The logic for converting 'YY' to 'YYYY' is implemented correctly, but ensure that the condition for determining the century (using 20 or 19) is consistently applied and tested with edge cases.